### ISSUE TITLE
Shell Command Agent executing commands inside Bundler context

### ISSUE THREAD

**Issue by lfcipriani** [cantino/huginn#1989]
Hi,

I have a ruby script that I want to execute from Shell Command Agent, however, since the command runs inside Bundler context, I'm not able to use the right gem version (google-api-client). I tried several ways of executing this out of Bundler scope but nothing worked so far.

However, Bundler has a [method to shell out](http://bundler.io/v1.3/man/bundle-exec.1.html#Shelling-out) when executing commands:

    Bundler.with_clean_env do
      `brew install wget`
    end

Is it possible to wrap the executing of Shell Command Agent with that method by default? Or at least give a config option on the agent?

Thanks

**Comment by knu** [comment]
Good point.  I remember when I needed this feature I just chose to unset unwanted environment variables in the shell command myself (like `unset RUBYOPT; unset BUNDLE_GEMFILE; PATH=$BUNDLER_ORIG_PATH; ...`), but I agree that it'd be cleaner to add a new option.

**Comment by knu** [comment]
I used `Bundle.with_original_env` in the above PR because `Bundle.with_clean_env` is obsolete, but the former reverts the environment variables to the state before bundler is setup, so if bundler is nested, you may not get a clean, bundler-free environment for running your command.  In that case, you need to clean up environment variables on your own as described in my previous comment.

**Comment by knu** [comment]
@lfcipriani Could you confirm if the above enhancement works for you?

**Comment by lfcipriani** [comment]
@knu I'll need to update my stack first. I'll let you know when I'm able to test. Thanks
